### ISSUE TITLE
10$: Report error on ()=>{}()

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#662]
The following code has a parse error:
```js
() => { console.log('hello'); }()
```
quick-lint-js accepts this code without error. It should instead report an error stating that parentheses are required around the expression. Here is the desired code:
```js
(() => { console.log('hello'); }())
```

**Comment by cgsdev0** [comment]
is the desired code here actually this?
```javascript
(() => { console.log('hello'); })()
```

I get an error when i paste the current code into my chrome console:
![image](https://user-images.githubusercontent.com/4583705/158919721-acb62634-ca8b-43b3-8a53-6e2a4bc5f0be.png)

**Comment by strager** [comment]
> is the desired code here actually this?

Yes.

**Comment by cgsdev0** [comment]
i'd like to claim this task!

I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.
